File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    /** solution
        1. 첫 째날 - progresses 배열을 돌면서 speeds 만큼 더한다.
        2. 첫 번째 shift 대상 검사해서 100이면 pop(), 그렇지 않으면 1 반복
        3, 첫 번째 대상 shift 할 수 있게 되면 shift 하고 그 이후의 것들도 검사.
    */

    while (progresses.length > 0) {
        
        const current = progresses.shift();
        const speed = speeds.shift();
        
        if (current >= 100) {
            answer[answer.length - 1] += 1;
            continue;
        }
        const days = Math.ceil((100 - current) / speed); // 작업 일수
        answer.push(1)
        
        for (let i=0;i<progresses.length;i++) {
            progresses[i] += (speeds[i] * days)
        }

        
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    
    # 앞에서 다 안 끝나면 뒤의 작업들도 미뤄뒀다 같이 배포함
    # ( (100 - progress) / speed ) + 1 (딱 떨어지지 않을 경우) 일 동안 작업
    # 각 작업 일수 계산 후, 7, 3 이면 2개의 기능이 배포됨.
    
    n = len(progresses)
    
    # 1. progresses 돌면서 작업 일수 계산
    days = []
    for i in range(n):
        progress = progresses[i]
        speed = speeds[i]
        
        days.append(math.ceil((100 - progress) / speed))
    
    # 2. 작업 일수 돌면서 각 배포에 몇 개의 기능이 배포 가능한지 합산

    maxDay = days[0]
    cnt = 1
    for j in range(1, n):

        if maxDay >= days[j]:
            # 이번 배포에 포함
            cnt += 1
            
        else:
            maxDay = days[j]
            answer.append(cnt)
            cnt = 1
    
    answer.append(cnt)
            
    return answer